src/components/SquareBox/SquareBox.tsx

```diff
@@ -1,57 +1,75 @@
+/* eslint-disable linebreak-style */
 import React from 'react';
 import { Button } from '@mui/material';
 import { StyleSheet, css } from 'aphrodite';
-import { Square } from './Square';
 import { COLORS } from '../../constants/colors';
 import { randomElement } from '../../functions/randomElement';
+import { Circle } from './Circle';
+import { Square } from './Square';
 
 const styles = StyleSheet.create({
-    main: {
-        display: 'flex',
-        flexDirection: 'column',
-        margin: '25px'
-    },
-    control: {
-        width: '100%',
-        display: 'flex',
+  main: {
+    display: 'flex',
+    flexDirection: 'column',
+    margin: '25px',
+  },
+  control: {
+    width: '100%',
+    display: 'flex',
+  },
+  squares: {
+    marginTop: '25px',
+    display: 'flex',
+    child: {
+      marginRight: '10px',
     },
-    squares: {
-        marginTop: '25px',
-        display: 'flex',
-        'child': {
-            marginRight: '10px'
-        }
-    },
-    square: {
-        marginRight: '15px'
-    }
+  },
+  square: {
+    marginRight: '15px',
+  },
 });
 
 export function SquareBox() {
-    const id = React.useId();
-    const [squares, setSquares] = React.useState<Array<string>>([]);
-
-    const deleteSquare = function () {
-        squares.pop();
-        setSquares(squares => [...squares]);
-    };
+  interface Shape {
+    color: string;
+    shape: string;
+  }
```

Comment:
Move this interface outside of SquareBox function, the body of SquareBox is re-run every time when the view has to be refreshed, so this interface is reacted over and over again.